src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspaceManager.cs

```diff
@@ -16,31 +16,54 @@
-    /// In future it should support workspaces of all kinds.
+    /// Manages remote workspaces. Currently supports only a single, primary workspace of kind <see
+    /// cref="WorkspaceKind.RemoteWorkspace"/>. In future it should support workspaces of all kinds.
     /// </summary>
     internal class RemoteWorkspaceManager
     {
-        /// <summary>
-        /// Default workspace manager used by the product. Tests may specify a custom <see cref="RemoteWorkspaceManager"/>
-        /// in order to override workspace services.
-        /// </summary>
-        internal static readonly RemoteWorkspaceManager Default = new RemoteWorkspaceManager(
-            new SolutionAssetCache(cleanupInterval: TimeSpan.FromMinutes(1), purgeAfter: TimeSpan.FromMinutes(3), gcAfter: TimeSpan.FromMinutes(5)));
-
         internal static readonly ImmutableArray<Assembly> RemoteHostAssemblies =
             MefHostServices.DefaultAssemblies
                 .Add(typeof(AspNetCoreEmbeddedLanguageClassifier).Assembly)
                 .Add(typeof(BrokeredServiceBase).Assembly)
                 .Add(typeof(RemoteWorkspacesResources).Assembly);
 
-        private readonly Lazy<RemoteWorkspace> _lazyPrimaryWorkspace;
+        /// <summary>
+        /// Default workspace manager used by the product. Tests may specify a custom <see
+        /// cref="RemoteWorkspaceManager"/> in order to override workspace services.
+        /// </summary>
+        /// <remarks>
+        /// The general thinking behind these timings is that we don't want to be too aggressive constantly waking up
+        /// and cleaning purging items from the cache.  But we also don't want to wait an excessive amount of time,
+        /// allowing it to get too full.
+        /// <list type="bullet">
+        /// <item>CleanupInterval=20s gives what feels to be a reasonable non-aggressive amount of time to let the cache
+        /// do its job, while also making sure several times a minute it is scanned for things that can be
+        /// dropped.</item>
+        /// <item>PurgeAfter=1m effectively states that an item will be dumped from the cache if not used in the last
+        /// minute.  This seems reasonable for keeping around all the parts of the current solutions in use, while
+        /// allowing values from teh past, or values removed from the solution to not persist too long.</item>
+        /// <item>GcAfter=1m means that we'll force some GCs to happen after that amount of time of *non-activity*.  In
+        /// other words, as long as OOP is being touched for operations, we will avoid doing the GCs.
+        /// </item>
+        /// </list>
+        /// </remarks>
+        internal static readonly RemoteWorkspaceManager Default = new(
+            workspace => new SolutionAssetCache(workspace, cleanupInterval: TimeSpan.FromSeconds(20), purgeAfter: TimeSpan.FromMinutes(1), gcAfter: TimeSpan.FromMinutes(1)));
```

Comment:
the workspace is now passed into teh SolutionAssetCache.  This ensures that the cache will not free up any items that hte workspace is current pointing to.